src/base/bittorrent/session.cpp

```diff
@@ -4475,10 +4481,20 @@ void Session::handleAddTorrentAlert(const lt::add_torrent_alert *p)
 {
     if (p->error)
     {
-        qDebug("/!\\ Error: Failed to add torrent!");
-        QString msg = QString::fromStdString(p->message());
-        LogMsg(tr("Couldn't load torrent. Reason: %1").arg(msg), Log::WARNING);
+        const QString msg = QString::fromStdString(p->message());
+        LogMsg(tr("Couldn't load torrent. Reason: %1.").arg(msg), Log::WARNING);
         emit loadTorrentFailed(msg);
+
+        const lt::add_torrent_params &params = p->params;
+        const bool hasMetadata = (params.ti && params.ti->is_valid());
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+        const auto id = TorrentID::fromInfoHash(hasMetadata ? params.ti->info_hashes() : params.info_hashes);
+#else
+        const auto id = TorrentID::fromInfoHash(hasMetadata ? params.ti->info_hash() : params.info_hash);
+#endif
+        const auto loadingTorrentsIter = m_loadingTorrents.constFind(id);
+        if (loadingTorrentsIter != m_loadingTorrents.cend())
+            m_loadingTorrents.erase(loadingTorrentsIter);
```

Comment:
Seems you can just remove it without checking: `loadingTorrentsIter.remove(id);`

Reply:
> 
> 
> Seems you can just remove it without checking: `m_loadingTorrents.remove(id);`

👍 
Done!